org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/AxesMetadataExample.java

```diff
@@ -79,149 +91,272 @@
-							needInterp = false;
+							throw new IllegalArgumentException(
+									DUPLICATE_INDEX_VALUES_NOT_ALLOWED);
 						}
 						else
 						{
-							// values don't match, need to interp
-							needInterp = true;
-						}
+							// check they're in the same direction
+							if (aMono != bMono)
+							{
+								throw new IllegalArgumentException(
+										AXES_MUST_ASCEND_IN_SAME_DIRECTION);
+							}
 
-						if (needInterp)
-						{
-							// we now need to decide which set of indices to use as the master
-							final boolean useAforIndexMaster;
+							// ok, they're both monotonic, in the same direction. we can continue.
+						}
 
-							// find the data limits
-							final double aMin = aIndices.min().doubleValue();
-							final double aMax = aIndices.max().doubleValue();
+						// find the data limits
+						final double aMin = aIndices.min().doubleValue();
+						final double aMax = aIndices.max().doubleValue();
+						final double bMin = bIndices.min().doubleValue();
+						final double bMax = bIndices.max().doubleValue();
 
-							final double bMin = bIndices.min().doubleValue();
-							final double bMax = bIndices.max().doubleValue();
+						if (aMax < bMin || aMin > bMax)
+						{
+							// datasets don't overlap
+							throw new IllegalArgumentException(INDICES_DO_NOT_OVERLAP);
+						}
+						else if (aMin <= bMin && aMax >= bMax)
+						{
+							// ok, b is wholly contained within A. Just generate
+							// points in A at the time in B
+							final Dataset interpolatedValues =
+									Maths.interpolate(aIndices, da, bIndices, null, null);
+							// remember the output axes, since we'll put them
+							// into the results
+							interpolatedValues.addMetadata(axesB);
+
+							operandA = db;
+							operandB = interpolatedValues;
+						}
+						else if (bMin < aMin && bMax > aMax)
+						{
+							// ok, A is wholly contained within B. Just generate
+							// points in B at the time in A
+							final Dataset interpolatedValues =
+									Maths.interpolate(bIndices, db, aIndices, null, null);
+							// remember the output axes, since we'll put them
+							// into the results
+							interpolatedValues.addMetadata(axesA);
 
-							if (aMax < bMin || aMin > bMax)
-							{
-								// datasets don't overlap
-								throw new IllegalArgumentException(
-										"The indices of the dataset do not overlap");
-							}
-							else if (aMin <= bMin && aMax >= bMax)
+							operandA = da;
+							operandB = interpolatedValues;
+						}
+						else
+						{
+							// ok, one isn't contained within the other,
+							// they merely overlap. Find the overlapping period
+							final double startPoint = aMin < bMin ? bMin : aMin;
+							final double endPoint = aMax > bMax ? bMax : aMax;
+
+							// to find the last matching entry, we have to work through
+							// the dataset in reverse.
+							final Dataset aReverse = aIndices.getSlice(null, null, new int[]
```

Comment:
Peter, switching to ````getSliceView()```` broke my tests,  so I've done some digging.

In the dataset returned from ````getSliceView```` the orders of the values haven't been reversed (they're still stored in ascending order).  The reversal is implemented via provision of a reversing iterator.

This trips over the call to ````DatasetUtils.getIndexLessThanOrEqualTo(d, o)````. I think that method relies on an ascending iterator index - it's performing the test against the correct sequence of elements, but the indexes it is using (and thus returning) are reversed.

Does this mean that if the product of  ````getSliceView()```` is to be passed to ````getIndexLessThanOrEqualTo()````, it first needs to be wrapped in a new DoubleDataset() to implement the reversal, this performing a copy operation.

I can provide a breaking JUnit test, if you think that is of value.  Or, I can just wrap my view in a new DoubleDataset.  Which would you prefer?